app/settings/surveys/survey-editor.directive.js

```diff
@@ -576,7 +584,7 @@ function SurveyEditorController(
         $scope.saving_survey = true;
         // Save the survey
         FormEndpoint
-        .saveCache($scope.survey)
+        .saveCache($scope.survey, $scope.createNew)
```

Comment:
You don't need the createNew attribute, simply delete the id from $scope.survey after deleting the form instead.